pkg/controller/policyrecommendation/policyrecommendation_controller_test.go

```diff
@@ -0,0 +1,310 @@
+		Expect(c.Create(ctx, kiibanaTLS.Secret(common.OperatorNamespace()))).NotTo(HaveOccurred())
+
+		Expect(c.Create(ctx, relasticsearch.NewClusterConfig("cluster", 1, 1, 1).ConfigMap())).NotTo(HaveOccurred())
+		Expect(c.Create(ctx, rtest.CreateCertSecret(render.ElasticsearchPolicyRecommendationUserSecret, common.OperatorNamespace(), render.GuardianSecretName)))
+		Expect(c.Create(ctx, rtest.CreateCertSecret(render.ElasticsearchADJobUserSecret, common.OperatorNamespace(), render.GuardianSecretName)))
+		Expect(c.Create(ctx, rtest.CreateCertSecret(render.ElasticsearchPerformanceHotspotsUserSecret, common.OperatorNamespace(), render.GuardianSecretName)))
+		Expect(c.Create(ctx, &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      render.ECKLicenseConfigMapName,
+				Namespace: render.ECKOperatorNamespace,
+			},
+			Data: map[string]string{"eck_license_level": string(render.ElasticsearchLicenseTypeEnterpriseTrial)},
+		})).NotTo(HaveOccurred())
+
+		Expect(c.Create(ctx, &v3.DeepPacketInspection{ObjectMeta: metav1.ObjectMeta{Name: "test-dpi", Namespace: "test-dpi-ns"}})).ShouldNot(HaveOccurred())
+
+		// Apply the policyrecommendation CR to the fake cluster.
+		Expect(c.Create(ctx, &operatorv1.PolicyRecommendation{ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"}})).NotTo(HaveOccurred())
+
+		// mark that the watches were successful
+		r.licenseAPIReady.MarkAsReady()
+		r.tierWatchReady.MarkAsReady()
+	})
+
+	Context("image reconciliation", func() {
+		It("should use builtin images", func() {
+			_, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// TODO(dimitrin): Add test back
+			// d := appsv1.Deployment{
+			// 	TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "v1"},
+			// 	ObjectMeta: metav1.ObjectMeta{
+			// 		Name:      render.PolicyRecommendationDeploymentName,
+			// 		Namespace: render.PolicyRecommendationNamespace,
+			// 	},
+			// }
+			// Expect(test.GetResource(c, &d)).To(BeNil())
+			// Expect(d.Spec.Template.Spec.Containers).To(HaveLen(1))
+			// controller := test.GetContainer(d.Spec.Template.Spec.Containers, "policy-recommendation-controller")
+			// Expect(controller).ToNot(BeNil())
+			// Expect(controller.Image).To(Equal(
+			// 	fmt.Sprintf("some.registry.org/%s:%s",
+			// 		components.ComponentPolicyRecommendation.Image,
+			// 		components.ComponentPolicyRecommendation.Version)))
+		})
+
+		It("should use images from imageset", func() {
+			Expect(c.Create(ctx, &operatorv1.ImageSet{
+				ObjectMeta: metav1.ObjectMeta{Name: "enterprise-" + components.EnterpriseRelease},
+				Spec: operatorv1.ImageSetSpec{
+					Images: []operatorv1.Image{
+						{Image: "tigera/policy-recommendation", Digest: "sha256:policyrecommendationcontrollerhash"},
+					},
+				},
+			})).ToNot(HaveOccurred())
+
+			_, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// TODO(dimitrin): Add test back
```

Comment:
Do we need to add this test back as part of this PR? 

If not, we should delete the commented code.